drivers/include/modbus_rtu.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2020-2023 Denis Litvinov <li.denis.iv@gmail.com>
+ * Copyright (C) 2023-2024 Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    drivers_modbus_rtu Modbus RTU
+ * @ingroup     drivers
+ * @brief       Modbus RTU
+ *
+ * Modbus RTU implementation with support for master and slave modes.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Modbus RTU interface definitions
+ *
+ * @author      Denis Litvinov <li.denis.iv@gmail.com>
+ * @author      Bas Stottelaar <bas.stottelaar@gmail.com>
+ */
+
+#ifndef MODBUS_RTU_H
+#define MODBUS_RTU_H
+
+#include "mutex.h"
+#include "periph/gpio.h"
+#include "periph/uart.h"
+#include "thread.h"
+
+#include "modbus.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Modbus RTU packet sizes
+ */
+#define MODBUS_RTU_PACKET_CRC_SIZE              2       /**< size of CRC */
+#define MODBUS_RTU_PACKET_REQUEST_SIZE_MIN      8       /**< minimal request size (with CRC) */
+#define MODBUS_RTU_PACKET_RESPONSE_SIZE_MIN     5       /**< minimal response size (with CRC) */
+#define MODBUS_RTU_PACKET_SIZE_MAX              256     /**< maximum response size (with CRC) */
+
+/**
+ * @brief   Modbus RTU device parameters
+ */
+typedef struct {
+    gpio_t pin_rts;         /**< RTS pin, @p GPIO_UNDEF if not used */
+    int pin_rts_enable;     /**< RTS pin state when transmitting */
+    uart_t uart;            /**< UART device */
+    uint32_t baudrate;      /**< UART Baudrate */
+} modbus_rtu_params_t;
+
+/**
+ * @brief   Modbus RTU device structure
+ */
+typedef struct {
+    modbus_t dev;                               /**< @ref modbus_t base class */
+    const modbus_rtu_params_t *params;          /**< device parameters */
+    uint32_t timeout;                           /**< amount of time (usec) to wait for a slave to
+                                                     begin sending */
+    uint32_t rx_timeout;                        /**< timeout between two bytes */
```

Comment:
can u make them a little more descriptive 

like "inter_fame_usec_min" and "byte_usec_max" 

is that correct?

maybe its is more like frame_space_rx (1,5 chars ) , frame_space_ tx (3,5 chars)

please check me on this

Reply:
I have moved `timeout` to the `params` section, and renamed `rx_timeout` to `byte_timeout`. 

rx_timeout is calculated during initialization, to save some computation in hot path (debatable).

Fields inside `modbus_rtu_t` are now marked as internal (with a comment).